src/features/recs-pathways/widgets.ts

```diff
@@ -0,0 +1,87 @@
+import { getConfig } from '../../shared/config';
+import { WIDGET_ENDPOINT_PROD } from '../../shared/constants';
+import { buildApiUrl } from '../../shared/url-builders';
+import type {
+  CategoryWidgetOptions,
+  GlobalWidgetOptions,
+  ItemWidgetOptions,
+  KeywordWidgetOptions,
+  PersonalizedWidgetOptions,
+  RecentlyViewedWidgetOptions,
+} from './widget-options.type';
+import type { WidgetResponse } from './widget-response.type';
+
+export async function getGlobalWidget(id: string, params: GlobalWidgetOptions): Promise<WidgetResponse> {
+  const { widgetEndpoint, ...config } = getConfig();
+  const defaults = {};
+
+  const queryParams = Object.assign(config, defaults, params);
+
+  const url = buildApiUrl(`${widgetEndpoint || WIDGET_ENDPOINT_PROD}/global/${id}`, queryParams);
+  return fetch(url).then((data) => data.json());
```

Comment:
Any reason for not to using `await` in this `async` function, but to use promise APIs directly?

Reply:
Not really, its a oneliner now vs 2 lines. I dont feel strongly about it.

Reply:
Maybe we can just remove `async` from the function definition then?  It feels a little strange to see no `await` in `async`, even though yes this perfectly functional.

Reply:
Good point, hm I think I will actually make it use async await syntax instead, in case want to do some processing later on with `data`